tests/OmniSharp.Roslyn.CSharp.Tests/CodeActionsWithOptionsFacts.cs

```diff
@@ -0,0 +1,323 @@
+                    public string Foo[||] { get; set; }
+                }
+
+                public class BaseClass {}
+";
+
+            var response = await FindRefactoringNamesAsync(code);
+
+            Assert.DoesNotContain("Pull 'Foo' up -> Pull members up to base type...", response);
+        }
+
+        [Fact]
+        public async Task Can_extract_interface()
+        {
+            const string code =
+                @"public class Class1[||]
+                {
+                    public string PropertyHere { get; set; }
+                }";
+
+            const string expected =
+                @"
+                public interface IClass1
+                {
+                    string PropertyHere { get; set; }
+                }
+
+                public class Class1 : IClass1
+                {
+                    public string PropertyHere { get; set; }
+                }
+                ";
+            var response = await RunRefactoringAsync(code, "Extract Interface...");
+
+            var foo = ((ModifiedFileResponse)response.Changes.First()).Buffer;
+            AssertIgnoringIndent(expected, ((ModifiedFileResponse)response.Changes.First()).Buffer);
+        }
+
+        [Fact]
+        public void Check_better_alternative_available_for_codeaction_with_options()
+        {
+            // This keeps record is Microsoft.CodeAnalysis.PickMembers.IPickMembersService public or not
+            // and should fail on case where interface is exposed. Which likely means that this is officially
+            // supported scenario by roslyn.
+            //
+            // In case it's exposed by roslyn team these services can be simplified.
+            // Steps are likely following:
+            // - Remove proxies
+            // - Replace ExportWorkspaceServiceFactoryWithAssemblyQualifiedName with Microsoft.CodeAnalysis.Host.Mef.ExportWorkspaceServiceAttribute
+            // - Fix proxy classes to implement IPickMembersService / IExtractInterfaceOptionsService ... instead of proxy and reflection.
+            // - Remove all factories using ExportWorkspaceServiceFactoryWithAssemblyQualifiedName and factory itself.
+            // Following issue may have additional information: https://github.com/dotnet/roslyn/issues/33277
+            var pickMemberServiceType = Assembly.Load("Microsoft.CodeAnalysis.Features").GetType("Microsoft.CodeAnalysis.PickMembers.IPickMembersService");
+            Assert.False(pickMemberServiceType.IsPublic);
+        }
+
+        private static void AssertIgnoringIndent(string expected, string actual)
+        {
+            Assert.Equal(TrimLines(expected), TrimLines(actual), false, true, true);
+        }
+
+        private static string TrimLines(string source)
+        {
+            return string.Join("", source.Split('\n').Select(s => s.Trim()));
+        }
+
+        private async Task<RunCodeActionResponse> RunRefactoringAsync(string code, string refactoringName, bool wantsChanges = false)
```

Comment:
this code looks identical to the code in `CodeActionsV2Facts`.
Could you make a new `AbstractCodeActionsTestFixture`, move that code there and use it in both of these test classes so that it's not duplicated?